torch/lib/THPP/tensors/generic/THTensor.cpp

```diff
@@ -50,6 +69,26 @@ auto THTensor<real>::contiguous() const -> std::unique_ptr<Tensor> {
 }
 
 template<>
+auto THTensor<real>::newSelect(int dimension, long sliceIndex) const -> THTensor* {
+  return new THTensor(THTensor_(newSelect(tensor, dimension, sliceIndex)));
```

Comment:
I think you haven't closed the brace (after `newSelect`). Same applies to the functions below.